src/Api/Order.php

```diff
@@ -129,13 +130,21 @@ public static function place_order( \WP_REST_Request $request ) {
 		$order_id = $checkout->create_order();
 		$order = new \WC_Order( $order_id );
 
-		do_action( Hooks::AFTER_ORDER, $request, $cart );
-
 		// If the user is not logged in, we need to pass the billing and shipping address to the order.
 		if ( ! is_user_logged_in() ) {
 			$order = self::update_guest_order( $request, $order );
+
+			if ( is_wp_error( $order ) ) {
```

Comment:
We also need to make sure the `$order_id` is from an order since with this code I can send a `page_id` or `post_id` as the `order_id` and is going to delete that.